gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dframe_ttt2.lua

```diff
@@ -0,0 +1,401 @@
+-- @realm client
+function PANEL:ClearFrame(w, h, title)
+	if isfunction(self.OnClear) then
+		local shouldCancel = self:OnClear() == false
+
+		if shouldCancel then return end
+	end
+
+	local oldW, oldH = self:GetSize()
+	local oldTitle = self:GetTitle()
+
+	self:Clear()
+	self:InitButtons()
+	self:ShowBackButton(false)
+
+	if (w and w ~= oldW) or (h and h ~= oldH) then
+		self:SetSize(w or oldW, h or oldH)
+		self:Center()
+	end
+
+	if title ~= oldTitle then
+		self:SetTitle(title or oldTitle)
+	end
+end
+
+---
+-- Closes the frame.
+-- @realm client
+function PANEL:CloseFrame()
+	self:SetDeleteOnClose(true)
+	self:Close()
+end
+
+---
+-- Function that is called when the frame is hidden, return false to cancel event.
+-- @return boolean Return false to cancel this event
+-- @hook
+-- @realm client
+function PANEL:OnHide()
+
+end
+
+---
+-- Function that is called when the frame is unhidden, return false to cancel event.
+-- @return boolean Return false to cancel this event
+-- @hook
+-- @realm client
+function PANEL:OnShow()
+
+end
+
+---
+-- Function that is called when the frame is rebuild
+-- @hook
+-- @realm client
+function PANEL:OnRebuild()
+
+end
+
+---
+-- Function that is called when the frame is about to be cleared, return false to cancel event.
+-- @return boolean Return false to cancel this event
+-- @hook
+-- @realm client
+function PANEL:OnClear()
+
+end
+
+function PANEL:InitButtons()
+	-- add close button
+	self.btnClose = vgui.Create("DButton", self)
+	self.btnClose:SetText("")
+	self.btnClose:SetVisible(true)
+
+	self.btnClose.DoClick = function(button)
+		self:Close()
+	end
+
+	self.btnClose.Paint = function(panel, w, h)
+		derma.SkinHook("Paint", "WindowCloseButton", panel, w, h)
+	end
+
+	-- add back button
+	self.btnBack = vgui.Create("DButton", self)
+	self.btnBack:SetText("")
+	self.btnBack:SetVisible(true)
+
+	self.btnBack.DoClick = function(button)
+
+	end
+
+	self.btnBack.Paint = function(panel, w, h)
+		derma.SkinHook("Paint", "WindowBackButton", panel, w, h)
+	end
+end
+
+function PANEL:SetTitle(strTitle)
+	self.title.text = strTitle
+end
+
+function PANEL:GetTitle()
+	return self.title.text
+end
+
+function PANEL:SetTitleFont(fontName)
+	self.title.font = fontName
+end
+
+function PANEL:GetTitleFont()
+	return self.title.font
+end
+
+function PANEL:SetPadding(left, top, right, bottom)
+	self.padding = {
+		left = left,
+		top = top,
+		right = right,
+		bottom = bottom
+	}
+
+	self:UpdatePadding()
+end
+
+function PANEL:UpdatePadding()
+	self:DockPadding(self.padding.left, vskin.GetHeaderHeight() + vskin.GetBorderSize() + self.padding.top, self.padding.right, self.padding.bottom)
+end
+
+function PANEL:ShowCloseButton(bShow)
+	self.btnClose:SetVisible(bShow)
+end
+
+function PANEL:ShowBackButton(bShow)
+	self.btnBack:SetVisible(bShow)
+end
+
+function PANEL:RegisterBackFunction(fn)
+	self.btnBack.DoClick = fn
+end
+
+function PANEL:Close()
+	self:SetVisible(false)
+
+	if self:GetDeleteOnClose() then
```

Comment:
Wouldn't `IsDeleteOnClose` matching better (boolean-based functions using `Is` in TTT2)

Reply:
Since this function is created by the gmod engine, I can not change the name.